pkg/workloads/spark_job/spark_util.py

```diff
@@ -435,34 +465,63 @@ def _transform(*values):
 
     if validate:
         transformed_column = ctx.transformed_columns[column_name]
+        columnType = ctx.get_inferred_column_type(column_name)
 
         def _transform_and_validate(*values):
             result = _transform(*values)
-            if not util.validate_column_type(result, transformed_column["type"]):
+            if not util.validate_column_type(result, columnType):
                 raise UserException(
                     "transformed column " + column_name,
                     "tranformation " + transformed_column["transformer"],
-                    "type of {} is not {}".format(result, transformed_column["type"]),
+                    "type of {} is not {}".format(result, columnType),
                 )
 
             return result
 
         transform_python_func = _transform_and_validate
 
-    column_data_type_str = ctx.transformed_columns[column_name]["type"]
+    column_data_type_str = ctx.get_inferred_column_type(column_name)
     transform_udf = F.udf(transform_python_func, CORTEX_TYPE_TO_SPARK_TYPE[column_data_type_str])
     return df.withColumn(column_name, transform_udf(*required_columns_sorted))
 
 
-def validate_transformer(column_name, df, ctx, spark):
+def validate_transformer(column_name, test_df, ctx, spark):
     transformed_column = ctx.transformed_columns[column_name]
 
     trans_impl, _ = ctx.get_transformer_impl(column_name)
 
     if hasattr(trans_impl, "transform_python"):
+        if transformed_column.get("transformer_path", None):
+            sample_df = test_df.collect()
+            sample = sample_df[0]
+            inputs = ctx.create_column_inputs_map(sample, column_name)
+            _, impl_args = extract_inputs(column_name, ctx)
+            transformedSample = trans_impl.transform_python(inputs, impl_args)
+            rowType = type(transformedSample)
+            isList = rowType == list
+
+            for row in sample_df:
+                inputs = ctx.create_column_inputs_map(row, column_name)
+                transformedSample = trans_impl.transform_python(inputs, impl_args)
+                if rowType != type(transformedSample):
+                    raise UserRuntimeException(
+                        "transformed column " + column_name,
+                        "type inference failed, mixed data types in dataframe.",
+                    )
+
+            typeConversionDict = PYTHON_TYPE_TO_CORTEX_TYPE
+            if isList:
```

Comment:
I would refactor out the code that infers the cortex type of the python object to a function. That way we can thoroughly unit test that function